src/function/algebra/simplify.js

```diff
@@ -403,36 +515,46 @@ export const createSimplify = /* #__PURE__ */ factory(name, dependencies, (
       const ruleType = typeof rule
       switch (ruleType) {
         case 'string':
-        {
-          const lr = rule.split('->')
-          if (lr.length === 2) {
-            rule = { l: lr[0], r: lr[1] }
-          } else {
-            throw SyntaxError('Could not parse rule: ' + rule)
-          }
-        }
+          rule = { s: rule }
         /* falls through */
         case 'object':
-          newRule = {
-            l: removeParens(parse(rule.l)),
-            r: removeParens(parse(rule.r))
+          newRule = {}
```

Comment:
Can you extract a little helper function to built up `newRule` here? Like `newRule = createNewRule(...)`? To not distract from the "main" logic here.

Reply:
Done. I called it "_canonicalizeRule" because it takes a rule object which might take various forms and produces a standard rule object that applyRule can deal with.